mk-poetry-dep.nix

```diff
@@ -129,7 +129,6 @@ pythonPackages.callPackage
         ++ lib.optional (stdenv.buildPlatform != stdenv.hostPlatform) pythonPackages.setuptools
         ++ lib.optional (!isSource) (getManyLinuxDeps fileInfo.name).pkg
         ++ lib.optional isDirectory buildSystemPkgs
-        ++ lib.optional (!__isBootstrap) pythonPackages.poetry
```

Comment:
This will break some downstreams (not all of the cases where this is required is covered by tests).
I would be open to something like `!__isBootstrap && !isCross` so that we can solve your issue while not breaking various projects using this code.

Reply:
Can you write down such a scenario? I'm happy to write one or more test cases, so that folks don't spend a ton of time tracking down the problem to this (like I did) without a test breaking.

Reply:
This is really the same as any other build-system missing caused by https://github.com/python-poetry/poetry/issues/2789.

Adding `poetry` to each build was just a hack to circumvent that breakage for stuff depending on Poetry which I figured would be a fairly cheap workaround considering you are very likely to have poetry in your dependency graph anyway if you use this tool.

Maybe as a minimal reproduction case you could try an environment with https://github.com/tweag/pynixutil in it?